rs/ledger_suite/icp/ledger/src/main.rs

```diff
@@ -812,6 +855,32 @@ fn pre_upgrade() {
         .expect("failed to write instructions consumed to stable memory");
 }
 
+#[cfg(feature = "upgrade-to-memory-manager")]
+#[export_name = "canister_pre_upgrade"]
+fn pre_upgrade() {
```

Comment:
According to the design [doc](https://docs.google.com/document/d/1v34PJ0sgYKIKKF2r4o_qhVaeSHqeUwhmR7XnuE5UB0o/edit#heading=h.mhhrcapko22j), the changes to the pre-upgrade hook should only happen in V2. With this change, it won't be possible to roll back to V0.